### DISCUSSION TITLE
error in running a simple test job using version 3.3.1

### DISCUSSION THREAD

**Discussion by LZ5278** [TRIQS/solid_dmft#87] [Q&A]
Dear Alex and all, 

I am trying to do an OS calculation with VASP PLO and Hubbard-I, then I run into some error.  
Need help to understand it. 

I am using Triqs/DFTTools 3.3.1, solid_dmft 3.3.1, Hubbard-I 3.3.0. And some other solvers all installed in version 3.3.x

The material being calculated is CeB6, non-magnetic, no SOC considered at the moment. 
It has the Ce-f bands crowded in [-2, +2]eV around Fermi level, at DFT level. 
My VASP input files are: [INCAR.zip](https://github.com/user-attachments/files/17271742/INCAR.zip), [KPOINTS.zip](https://github.com/user-attachments/files/17271743/KPOINTS.zip), [POSCAR.zip](https://github.com/user-attachments/files/17271744/POSCAR.zip), [POTCAR.zip](https://github.com/user-attachments/files/17271745/POTCAR.zip)

After the VASP run I copied these output files into another folder, 
[DOSCAR.zip](https://github.com/user-attachments/files/17271780/DOSCAR.zip), [EIGENVAL.zip](https://github.com/user-attachments/files/17271781/EIGENVAL.zip), [IBZKPT.zip](https://github.com/user-attachments/files/17271782/IBZKPT.zip), [LOCPROJ.zip](https://github.com/user-attachments/files/17271783/LOCPROJ.zip), [POSCAR.zip](https://github.com/user-attachments/files/17271784/POSCAR.zip), [PROJCAR.zip](https://github.com/user-attachments/files/17271785/PROJCAR.zip) 
and, created the plo.cfg, [plo.zip](https://github.com/user-attachments/files/17271811/plo.zip), in the same directory. 

Then I executed these commands for the converter step: 
```bash
from triqs_dft_tools.converters.vasp import *
import triqs_dft_tools.converters.plovasp.converter as plo_converter
plo_converter.generate_and_output_as_text('plo.cfg', vasp_dir='./')
Converter = VaspConverter(filename = 'vasp')
Converter.convert_dft_input() 
```
The converter step screen output is here: [screen_output.zip](https://github.com/user-attachments/files/17271913/screen_output.zip). 
I got the vasp.h5 [vasp.zip](https://github.com/user-attachments/files/17271961/vasp.zip), together with pdos_0_0.dat [pdos_0_0.zip](https://github.com/user-attachments/files/17271926/pdos_0_0.zip), vasp.ctrl [vasp (2).zip](https://github.com/user-attachments/files/17271970/vasp.2.zip), vasp.pg1 [vasp (3).zip](https://github.com/user-attachments/files/17271974/vasp.3.zip). 

At this point there's no error as I can see. 

Then I run solid_dmft in command line "mpirun solid_dmft dmft_config.toml" with the input file [dmft_config.zip](https://github.com/user-attachments/files/17272011/dmft_config.zip). 
It threw an error message: 
```bash
Starting run with 4 MPI rank(s) at : 2024-10-06 21:31:20.811671
Reading the config file dmft_config.toml
Traceback (most recent call last):
  File "/home/lzhang/triqs-3.3.1/install/bin/solid_dmft", line 45, in <module>
    run_solid_dmft.main(sys.argv)
  File "/home/lzhang/triqs-3.3.1/install/lib/python3.11/site-packages/solid_dmft/main.py", line 151, in main
    run_dmft(params, config_file_name)
  File "/home/lzhang/triqs-3.3.1/install/lib/python3.11/site-packages/solid_dmft/main.py", line 64, in run_dmft
    full_params = postproc_toml_dict.merge_config_with_default(params, default_params,
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lzhang/triqs-3.3.1/install/lib/python3.11/site-packages/solid_dmft/io_tools/postproc_toml_dict.py", line 173, in merge_config_with_default
    output = _apply_default_values(cfg_inp, cfg_def, match_key)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lzhang/triqs-3.3.1/install/lib/python3.11/site-packages/solid_dmft/io_tools/postproc_toml_dict.py", line 81, in _apply_default_values
    raise ValueError(f'Unknown keys {unknown_keys} found in section "{section_name}". '
ValueError: Unknown keys {'store_solver'} found in section "solver". All valid keys have to be in the default config.
``` 

I am not sure if this comes from somewhere in the input file dmft_config.toml. 
If so, then which line caused it? 
Or, is there anything that looks wrong in VASP or converter step?  

Btw, I have failed 2 tests during the installation of solid_dmft 3.3.1. But it looks _not_ related to the above error. 
```bash
The following tests FAILED:
         10 - test_plot_correlated_bands (Failed)
         19 - maxent_svo_hubbardI_basic (Failed)
Errors while running CTest
``` 
I had run "ctest -V -R failed_test" for these two, and the output is [here](https://github.com/user-attachments/files/17272308/failed_tests.zip), just in case they might be related.  
No error during the installation of Triqs library and DFTTools, and the tests there were all passed.  
 
Thanks a lot and appreciate any hint! 

Best, 
L. Zhang

**Comment by the-hampel** [comment]
Hi @LZ5278 ,

sorry for the delayed answer. 

The error you are seeing here is just that the flag `store_solver` is no longer part of each solver section, but rather is placed in the general section of the dmft_config.toml file. Please just move this flag up into the general section and try again. Otherwise I do not see an obvious problem here. 

The test failures you are observing are just related to the fact that you do not have a triqs/maxent installation on your system. You can fix this by simply installing maxent via pip: `pip install triqs-maxent`.

Best,
Alex

**Reply by LZ5278** [reply]
Hi Alex, 

I moved the flag "store_solver" up into the general section, and it runs. 
I just run 1 iteration with 1 mpi task for quick testing, [dmft_config.toml](https://github.com/user-attachments/files/17301775/dmft_config.zip). 
Well by the end it threw another error, 
```bash
Traceback (most recent call last):
  File "/home/lzhang/triqs-3.3.1/install/bin/solid_dmft", line 45, in <module>
    run_solid_dmft.main(sys.argv)
  File "/home/lzhang/triqs-3.3.1/install/lib/python3.11/site-packages/solid_dmft/main.py", line 151, in main
    run_dmft(params, config_file_name)
  File "/home/lzhang/triqs-3.3.1/install/lib/python3.11/site-packages/solid_dmft/main.py", line 131, in run_dmft
    dmft_cycle(general_params, solver_params, advanced_params,
  File "/home/lzhang/triqs-3.3.1/install/lib/python3.11/site-packages/solid_dmft/dmft_cycle.py", line 571, in dmft_cycle
    observables, is_converged) = _dmft_step(sum_k, solvers, it, general_params, solver_params, gw_params,
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lzhang/triqs-3.3.1/install/lib/python3.11/site-packages/solid_dmft/dmft_cycle.py", line 738, in _dmft_step
    solvers[icrsh].solve(it=it)
  File "/home/lzhang/triqs-3.3.1/install/lib/python3.11/site-packages/solid_dmft/dmft_tools/solver.py", line 498, in solve
    self.triqs_solver.solve(h_int=self.h_int, **self.triqs_solver_params)
  File "/home/lzhang/triqs-3.3.1/install/lib/python3.11/site-packages/triqs_hubbardI/solver.py", line 196, in solve
    self.G_w = atomic_g_w(self.ad, self.beta, self.gf_struct, (self.w_min,self.w_max), self.n_w, self.idelta)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Error: no suitable C++ overload found in implementation of function atom_diag.atomic_g_w

atomic_g_w(triqs::atom_diag::atom_diag<false> atom, double beta, gf_struct_t gf_struct, std::pair<double,double> energy_window, int n_w, double broadening, std::vector<std::pair<int,int>> excluded_states) -> block_gf<refreq>
 failed with the error :
  Python object is not a triqs.atom_diag.atom_diag.AtomDiagReal but a triqs.atom_diag.atom_diag.AtomDiagComplex

atomic_g_w(triqs::atom_diag::atom_diag<true> atom, double beta, gf_struct_t gf_struct, std::pair<double,double> energy_window, int n_w, double broadening, std::vector<std::pair<int,int>> excluded_states) -> block_gf<refreq>
 failed with the error :
  must be real number, not NoneType
```

How can I fix this? 

Thanks,
L. Zhang

**Reply by the-hampel** [reply]
Ah I think there might be actually a bug in solid_dmft. Can you try to add in the general section of the dmft_config.toml: `eta=0.01` and see if the error disappears? Thanks! If that is the case I need to to fix something in solid_dmft and I let you know what the problem is of course as well.

**Reply by LZ5278** [reply]
It makes it very slow. I don't know if it's expected. 
Without "eta=0.01", the one-iter one-task job took ~2.5 mins on my laptop, ended with the above error. 
The error popped after the chunk of "The local Hamiltonian of the problem: ................ ". (complete [screen_output](https://github.com/user-attachments/files/17319192/screen_output.zip)) 
While, with "eta=0.01", the job took 60 mins after "The local Hamiltonian of the problem:" was displayed, and it's not finishing. 
It looks like it's stuck there. I force stopped it. Assuming it's expected, I am trying an 8-task job now.

**Reply by the-hampel** [reply]
well I fear that for a 7 orbital problem the calculation of some of the green's functions might take some time. Unfortunately the HubbardI solver does not provide any MPI parallelism. So more ranks will not help here. Is there a way for you to maybe test this with fewer orbitals first? It might also be useful to limit the HubbardI solver to real dropping the small imaginary elements you have in your local Hamiltonian. Unfortunately this is not something that is broken in solid_dmft. This is some improvement we would need to add to triqs/HubbardI. Maybe try to let it run on a single core for longer and see if it eventually finishes...

**Reply by LZ5278** [reply]
I am trying to do this, will update asap.
